gateway/uplink.go

```diff
@@ -192,15 +194,15 @@ func executeDecoder(ctx context.Context, script string, vars map[string]interfac
 	select {
 	case <-timeoutCtx.Done():
 		vm.Interrupt <- func() {
-			errors.New("ctx canceled")
 		}
+		cancel()
```

Comment:
maybe can `defer cancel()` above this instead? unless there's a situation in which you wouldn't want to call cancel before this function returns